### PR TITLE
    WINDUP-984  Create an offline index of packages (or classes) in selected API artifacts

### PR DESCRIPTION
**Pull request by OndraZizka**
nexus-repository-indexer needs a release for this to work.
https://github.com/windup/nexus-repository-indexer/pull/4
https://github.com/windup/nexus-repository-indexer/pull/5

### CODE REVIEW

File: rules-java-archives/addon/src/main/java/org/jboss/windup/rules/apps/java/archives/identify/pkg/LucenePackageToArtefactMapper.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import org.jboss.windup.rules.apps.java.archives.identify.LuceneArchiveIdentificationService;
+import org.jboss.windup.util.Logging;
+import org.jboss.windup.util.exception.WindupException;
+
+/**
+ * Identifies archives by a contained package, using pre-created Lucene index. See the nexus-repository-indexer project.
+ *
+ * @author <a href="http://ondra.zizka.cz/">Ondrej Zizka, zizka@seznam.cz</a>
+ */
+public class LucenePackageToArtefactMapper extends LuceneArchiveIdentificationService
```
Comment (jsight) [inline rules-java-archives/addon/src/main/java/org/jboss/windup/rules/apps/java/archives/identify/pkg/LucenePackageToArtefactMapper.java]: Artifact not Artefact :)

Comment (OndraZizka) [inline rules-java-archives/addon/src/main/java/org/jboss/windup/rules/apps/java/archives/identify/pkg/LucenePackageToArtefactMapper.java]: Yeah I never know for sure :)

**Comment by OndraZizka** [comment]
Rebased, renamed, applied https://github.com/OndraZizka/windup/pull/3

**Comment by jsight** [comment]
This just needs to be part of https://github.com/windup/windup/pull/878, IMO.

**Comment by OndraZizka** [comment]
I will include this in the next iteration, after #878 is merged.

**Pull request closed without merging**